be/src/storage/persistent_index.cpp

```diff
@@ -3027,8 +3027,8 @@ Status PersistentIndex::_insert_rowsets(Tablet* tablet, std::vector<RowsetShared
 }
 
 bool PersistentIndex::_need_rebuild_index(const PersistentIndexMetaPB& index_meta) {
-    if (index_meta.format_version() != PERSISTENT_INDEX_VERSION_2 &&
-        index_meta.format_version() != PERSISTENT_INDEX_VERSION_3) {
+    if (index_meta.l0_meta().format_version() != PERSISTENT_INDEX_VERSION_2 &&
```

Comment:
We should downgrade from 2.5->2.4->2.3? Why direct from 2.5->2.3?

Reply:
> We should downgrade from 2.5->2.4->2.3? Why direct from 2.5->2.3?

Version 2.4 is not a LTS version, so many user will upgrade from version 2.3 to version 2.5. If they meet errors in version 2.5, they may downgrade to 2.3 directly.